Dockerfile

```diff
@@ -111,23 +101,27 @@ RUN apt-get update && \
     apt-get install -y --no-install-recommends --fix-missing \
     libgl1 \
     libglib2.0-0 \
-    python3-venv && \
+    python3-venv \
+    numactl && \
     apt-get clean && \
     rm -rf  /var/lib/apt/lists/*
 
-# PIP extra index url: THU tuna
-ENV PIP_EXTRA_INDEX_URL=https://pypi.tuna.tsinghua.edu.cn/simple
-
-ARG SDWEBUI_DEPS=https://raw.githubusercontent.com/vladmandic/automatic/c02ccc4f0001a8eb9b9b5ff32522f067bbe55103/requirements.txt
-ADD ${SDWEBUI_DEPS} /tmp/
-RUN --mount=type=cache,target=/root/.cache/pip \
-    pip install -r /tmp/requirements.txt && \
-    rm /tmp/requirements.txt
-
 COPY startup.sh /bin/
+COPY setperms.sh /bin/
+RUN chmod 755 /bin/startup.sh
+RUN chmod 755 /bin/setperms.sh
+
+# Set user and group
+ARG uid=1000
+ARG gid=1000
+RUN groupadd -g ${gid} sd-webui
+RUN useradd -u ${uid} -g sd-webui --no-log-init -m sd-webui
```

Comment:
Launching the web server via `webui.sh` as a non-root user would cause file permission issues on windows.
For example, if users copy a model from host to the bind-mounted folder that is shared with the container, the ownership of the copied model file is `root:root`. Then the Web UI would complain about permission issues when loading the model file.
And unfortunately, it seems we don't have other workaround than allowing the application (`webui.sh`) to run with the `root` user.
See [Permissions errors on data directories for shared volumes](https://docs.docker.com/desktop/troubleshoot/topics/#volumes)

Reply:
I didn't want to run as root user so i added these.
Adding -f as the first cmd argument allows to run as root user.